strings.go

```diff
@@ -5,30 +5,102 @@ import (
 	"fmt"
 	"go/ast"
 	"go/token"
+	"go/types"
 	"strconv"
 	"strings"
 
 	"golang.org/x/tools/go/ast/astutil"
 )
 
-func obfuscateLiterals(files []*ast.File) []*ast.File {
-	pre := func(cursor *astutil.Cursor) bool {
-		decl, ok := cursor.Node().(*ast.GenDecl)
-		if !ok || decl.Tok != token.CONST {
-			return true
+func isTypeAliasStr(typ types.Type) bool {
```

Comment:
a type alias in Go is very clearly defined; it's `type T1 = T2`. you don't mean that here, so please don't use the "alias" term anywhere.

also, I'm not sure that the logic is correct. What you want to do is find those string literals which could be converted to variables without breaking typechecking.

The only difference between a string constant and a string variable is, like we discussed in the other PR, whether or not it can be assigned to certain types, as per https://golang.org/ref/spec#Assignability.

So I think you want to use `return types.AssignableTo(strType, typ)` instead.

Also, I'm starting to think that this code gives up too easily. If we have `var v myString = "foo"`, we could convert the code to just be `var v myString = myString(str())`. `myString(str())` will work as long as `types.ConvertibleTo(strType, typ)` is true.

Reply:
It checks if the given type is an type alias of string therefore I think the name is right.

Reply:
I still think that's wrong - a "type alias of string" is literally identical to the string type:

    type myAlias = string

If you really wanted to do that, then you should just do `types.Identical`. But I assume that won't be super useful :)

Reply:
It's not identical to go/types it checks if it is exactly the same type.

Reply:
I mean, this depends on what your code intends to do. I just want to empphasize that "is a type alias of" is not what you're implementing here.

Reply:
You are right that it gives up a little to easy, would address that in a future pr however.

Reply:
Sure. The code would be pretty similar, and the test code would only need minor changes to check that more literals are garbled.

Reply:
I now realized that Convertible is not the right check for type aliases because many other types can be converted to string e.g.

`string(int) string([]bytes)`

this is currently not a big deal because we skip all types which are not equal to string but convertible anyway. If we want to do `mystring(encrypted())` we would need to rewrite this check.